src/Dscp/DB/DSL/Interpret/Sqlite3.hs

```diff
@@ -0,0 +1,192 @@
+
+{-# LANGUAGE QuasiQuotes #-}
+
+module Dscp.DB.DSL.Interpret.Sqlite3
+    ( ensureSchemaIsSetUp
+    ) where
+
+import Universum
+
+import qualified Data.Set as Set (Set, empty, member, singleton)
+import Data.Time.Clock (UTCTime)
+
+import Database.SQLite.Simple (Only (..))
+import Database.SQLite.Simple.Internal (Connection (..))
+import Database.SQLite3 (exec)
+
+import Text.InterpolatedString.Perl6 (q, qc, qq)
+
+import Dscp.Core.Types (Address (..), Assignment (..), AssignmentType (..), CourseId (..),
+                        Grade (..), SignedSubmission (..), StudentId, Submission (..),
+                        SubmissionSig, SubmissionType (..), SubmissionWitness (..))
+import Dscp.Crypto (PublicKey)
+import Dscp.DB.DSL.Class
+import Dscp.DB.DSL.Interpret.Sqlite3.Schema (schema)
+import Dscp.DB.DSL.Interpret.Sqlite3.Instances ()
+import Dscp.DB.SQLite
+import Dscp.Educator.Txs (PrivateTx (..), PrivateTxId)
+
+ensureSchemaIsSetUp :: Connection -> IO ()
+ensureSchemaIsSetUp (Connection db) = do
+    exec db schema
```

Comment:
In my world, `Dscp.DB.DSL.Interpret.*` modules are for logic directly related to processing of external search queries, encoded in the search query DSL. This function (together with database schema init query should go somewhere to `Dscp.DB.SQLite.*`. This is also true for `FromRow`/`ToRow` instances.

Reply:
Will move.